libcontainer/cgroups/fs/rdma.go

```diff
@@ -0,0 +1,116 @@
+// +build linux
+
+package fs
+
+import (
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+type RdmaGroup struct {
+}
+
+func (s *RdmaGroup) Name() string {
+	return "rdma"
+}
+
+func (s *RdmaGroup) Apply(path string, d *cgroupData) error {
+	return join(path, d.pid)
+}
+
+func createCmdString(device string, limits configs.LinuxRdma) string {
+	cmdString := device
+	if limits.HcaHandles != nil {
+		cmdString += " hca_handle=" + strconv.FormatUint(uint64(*limits.HcaHandles), 10)
+	}
+	if limits.HcaObjects != nil {
+		cmdString += " hca_object=" + strconv.FormatUint(uint64(*limits.HcaObjects), 10)
+	}
+	return cmdString
+}
+
+func (s *RdmaGroup) Set(path string, cgroup *configs.Cgroup) error {
+	for device, limits := range cgroup.Resources.Rdma {
+		if err := fscommon.WriteFile(path, "rdma.max", createCmdString(device, limits)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *RdmaGroup) GetStats(path string, stats *cgroups.Stats) error {
+	if !cgroups.PathExists(path) {
+		return nil
+	}
+	currentData, err := fscommon.ReadFile(path, "rdma.current")
+	if err != nil {
+		return err
+	}
+	currentPerDevices := strings.Split(currentData, "\n")
+	maxData, err := fscommon.ReadFile(path, "rdma.max")
+	if err != nil {
+		return err
+	}
+	maxPerDevices := strings.Split(maxData, "\n")
+	// If device got removed between reading two files, ignore returning
+	// stats.
+	if len(currentPerDevices) != len(maxPerDevices) {
+		return nil
+	}
+	currentEntries := ConvertRdmaEntry(currentPerDevices)
+	maxEntries := ConvertRdmaEntry(maxPerDevices)
+
+	stats.RdmaStats = cgroups.RdmaStats{
+		RdmaLimit:   maxEntries,
+		RdmaCurrent: currentEntries,
+	}
+
+	return nil
+}
+
+// Following is intentionally public as it going to be used by Rdma controller in fs2
+// ParseRdmaKV: parse raw string to RdmaEntry.
+func ParseRdmaKV(raw string, entry *cgroups.RdmaEntry) {
+	var value uint64
+	var err error
+
+	parts := strings.SplitN(raw, "=", 3)
+	if len(parts) == 2 {
+		if parts[1] == "max" {
+			value = math.MaxUint32
```

Comment:
`value` is declared as `uint64` but here you use `MaxUint32`. Why not `MaxUint64` (or declare `value` as `uint32`)?

Reply:
@kolyshkin oh makes sense i'll re-visit this entire PR once this gets merged as well https://github.com/opencontainers/runc/pull/3024 WDYT ?

Reply:
@kolyshkin So i remember reason for this was `strconv.ParseUInt` always returns uint64 event when `bitsize` is capped to `32`. So i had to cast back using `uint32`.

Reply:
Resolved.